src/librbd/cache/pwl/InitRequest.cc

```diff
@@ -148,11 +150,48 @@ void InitRequest<I>::handle_set_feature_bit(int r) {
     lderr(cct) << "failed to set feature bit: " << cpp_strerror(r)
                << dendl;
     save_result(r);
-  } else if (m_image_ctx.discard_granularity_bytes) {
-    ldout(cct, 1) << "RWL image cache is enabled and "
-                  << "set discard_granularity_bytes = 0." << dendl;
-    m_image_ctx.discard_granularity_bytes = 0;
+
+    shutdown_image_cache();
```

Comment:
Nit: just return after this call so that you don't need to create an `else` clause

Reply:
Done